elobot.py

```diff
@@ -11,195 +11,254 @@
-        loser_new_elo = round(loser.rating + loser.k_factor() * (0 - loser_expected_score))
+        winner_elo_delta = winner_new_elo - winner.rating
+        loser_elo_delta  = loser_new_elo  - loser.rating
 
-        winner_elo_delta, loser_elo_delta = abs(winner_new_elo - winner.rating), abs(loser_new_elo - loser.rating)
+        winner.wins += 1
+        loser.losses += 1
 
-        winner.wins, loser.losses = winner.wins + 1, loser.losses + 1
-        winner.rating, loser.rating = winner_new_elo, loser_new_elo
+        winner.rating = winner_new_elo
+        loser.rating = loser_new_elo
 
         return winner_elo_delta, loser_elo_delta
 
+    def apply_match(self, match):
+        """
+        Apply a match to the ranking system, changing winner and loser's elos.
+        Return (winner elo delta, loser elo delta)
+        """
+        if not match.pending:
+            raise ValuError("Match must be pending to apply.")
+
+        with db.transaction():
+            winner = self.players[match.winner_handle]
+            loser  = self.players[match.loser_handle]
+            winner_elo_delta, loser_elo_delta = self.rank_game(winner, loser)
+
+            match.pending = False
+            match.save()
+
+        return (winner_elo_delta, loser_elo_delta)
+
     def init_players(self):
-        print('initializing in-memory player objects')
-        print('loading match history')
+        """Initializes self.players with the games stored in the database"""
+        print('Initializing players...')
         matches = list(Match.select().order_by(Match.id))
-        print('simulating games...')
         for match in matches:
+            print('Recapping match: {}'.format(match.__dict__))
             if not match.pending:
-                winner, loser = self.players[match.winner_handle], self.players[match.loser_handle]
+                winner = self.players[match.winner_handle]
+                loser  = self.players[match.loser_handle]
                 self.rank_game(winner, loser)
-        print('finished initializing players')
-        print('players: {}'.format([ str(player) for player in self.players.values() ]))
-
+        print('Player initialization complete.')
 
-    def connect(self):
-        print('connecting!')
+    def ensure_connected(self):
         sleeptime = 0.1
-
-        while True:
+        while not self.slack_client.server.connected:
+            print('Was disconnected, attemping to reconnect...')
             try:
                 self.slack_client.rtm_connect()
-                break
-            except:
-                time.sleep(sleeptime)
-
-                #Exponential back off with a max wait of 30s
-                if sleeptime < 30:
-                    sleeptime *= 2
+            except:  # TODO: Except what
+                pass
+            time.sleep(sleeptime)
+            sleeptime = min(30, sleeptime * 2)  # Exponential back off with a max wait of 30s
```

Comment:
why are there two spaces in front of the comment

Reply:
PEP8 standard

Reply:
what the hell..